src/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -1018,32 +1048,38 @@ private static void AddValue(HeaderStoreItemInfo info, object value, StoreLocati
         // Since most of the time we just have 1 value, we don't create a List<object> for one value, but we change
         // the return type to 'object'. The caller has to deal with the return type (object vs. List<object>). This 
         // is to optimize the most common scenario where a header has only one value.
-        internal object GetParsedValues(string name)
+        internal object GetParsedValues(HeaderDescriptor descriptor)
         {
-            Debug.Assert((name != null) && (name.Length > 0));
-            Debug.Assert(HttpRuleParser.GetTokenLength(name, 0) == name.Length);
-
             HeaderStoreItemInfo info = null;
 
-            if (!TryGetAndParseHeaderInfo(name, out info))
+            if (!TryGetAndParseHeaderInfo(descriptor, out info))
             {
                 return null;
             }
 
             return info.ParsedValue;
         }
 
-        private void PrepareHeaderInfoForAdd(string name, out HeaderStoreItemInfo info, out bool addToStore)
+        // This is only used by unit tests.
```

Comment:
Is there any way we can enforce this?  Or change the unit tests to avoid it so we don't have to add this?  Seems like we could end up incurring some unexpected expenses accidentally if code ends up using it in the product.